src/resolvers/resolver.ts

```diff
@@ -20,6 +20,26 @@ const resolvers = {
       const newUser = await User.create({ email, password });
       return newUser;
     },
+    async updateUserRole(_: any, { id, role }: any) {
+      const userExists = await User.findOne({ id: id });
+      if (!userExists) throw new Error("User doesn't exist");
+         const updatedUser = await User.findOneAndUpdate(
+          {
+              _id: id
+          },
+          {
+              $set: {
+                 role: role
+          
+              }
+          }, { new: true }, (err, User) => {
+              if (err) {
```

Comment:
Since you are using the `await` in the call, please dont pass in the callback to avoid double database call.


And also you can update the `userExists` without finding the user again since we already have the user. 

The whole part can be easily done by 
```
await userExists.update({role}, {new: true}})
return userExists
```